spec/models/auth_policies/source_client_policy_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'rails_helper'
+
+RSpec.describe GrdaWarehouse::AuthPolicies::SourceClientPolicy, type: :model do
+  let(:data_source) { create :data_source_fixed_id }
+  let(:organization) { create :hud_organization, data_source: data_source }
+  let(:project) { create :grda_warehouse_hud_project, organization: organization }
+  let(:client) do
+    client = create(:hud_client, data_source: data_source)
+    create :hud_enrollment, client: client, data_source: data_source, project: project
+    client
+  end
+
+  # Permissions that will be granted through the role
+  let(:permissions) do
+    {
+      can_view_client_name: true,
+      can_view_client_photo: true,
+      can_view_full_dob: true,
+      can_view_full_ssn: false,
+      can_view_hiv_status: nil,
+    }
+  end
+
+  let(:role) { create(:role, **permissions.compact_blank) }
+
+  shared_examples 'permission checks' do |role_present|
+    it 'handles basic permissions appropriately' do
+      [
+        [:can_view_name?, true],
+        [:can_view_photo?, true],
+        [:can_view_full_dob?, true],
+        [:can_view_full_ssn?, false],
+        [:can_view_hiv_status?, false],
+      ].each do |method, expected|
+        expected = role_present ? expected : false
+        actual = policy.send(method)
+        expect(actual).to eq(expected), "#{method}: #{actual} != #{expected}"
+      end
+    end
+  end
+
+  shared_examples 'roi checks' do
+    context 'with ROI authorizations' do
+      before(:each) do
+        create(:warehouse_client, source: client, data_source: data_source)
+        create(:client_roi_authorization, destination_client: client.destination_client)
+        role.update!(can_search_clients_with_roi: true, can_view_client_enrollments_with_roi: true)
+      end
+
+      it 'grants view and search permissions through ROI' do
+        expect(policy.can_view?).to be true
+        # expect(policy.can_search?).to be true
+      end
```

Comment:
Should we clean these up?  I think we decided to implement the search permission only through a scope at this point.